### PR TITLE
doc: release notes: Add driver and sensor changes for 1.11 release

### PR DESCRIPTION
**Pull request by MaureenHelm**
Fills in the drivers subsection with a list of major additions and
changes in the 1.11 release.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

**Comment by pfalcon** [comment]
Thanks for @MaureenHelm going for this. Some comments:

* I'm not sure if there would be comments on capitalization of "i2c", "pwm", etc.
* I'd suggest to remove "Redefined SENSOR_CHAN_DISTANCE to be in meters" - it was added with the vl53l0x driver which was merged in 1.11, so no user-visible changes here.
* Otherwise, I don't know if we want to mark more explicitly that "Redefined SENSOR_CHAN_HUMIDITY in percents" is a backward incompatible change.

Thanks!

**Review by dbkinder — APPROVED** [review]

File: doc/release-notes-1.11.rst [hunk 1]
```diff
@@ -59,6 +59,16 @@ Boards
...
+* Added i2c master, qspi flash, and gpio drivers for nios-II
+* Added pinmux, gpio, serial drivers for lpc54114
+* Added pinmux, gpio, serial, spi, and watchdog drivers for sam0
+* Added apa102 and ws2821b led_strip drivers
+* Added pwm led driver for esp32
+* Added native entropy driver
+* Moved some sensors to dts
+* Added amg88xx, ccs811, and vl53l0x sensor drivers
+* Redefined SENSOR_CHAN_DISTANCE to be in meters
+* Redefined SENSOR_CHAN_HUMIDITY in percents
```
Comment (dbkinder) [inline doc/release-notes-1.11.rst]: s/percents/percent/

**Review by dbkinder — CHANGES_REQUESTED** [review]
I'll make a pass to fix capitalization before we release.

**Comment by MaureenHelm** [comment]
- Rebased and fixed merge conflict
- Removed "Redefined SENSOR_CHAN_DISTANCE to be in meters"
- Replaced "percents" with "percent"

**Review by dbkinder — APPROVED** [review]
thanks.

**Pull request merged**